src/Microsoft.PowerShell.Commands.Management/commands/management/Navigation.cs

```diff
@@ -3603,22 +3603,88 @@ protected override bool ProviderSupportsShouldProcess
 
         #region Command code
 
+        private Collection<PathInfo> GetResolvedPaths(string path)
+        {
+            Collection<PathInfo> results = new Collection<PathInfo>();
+            try
+            {
+                results = SessionState.Path.GetResolvedPSPathFromPSPath(path, CmdletProviderContext);
+            }
+            catch (PSNotSupportedException notSupported)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        notSupported.ErrorRecord,
+                        notSupported));
+            }
+            catch (DriveNotFoundException driveNotFound)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        driveNotFound.ErrorRecord,
+                        driveNotFound));
+            }
+            catch (ProviderNotFoundException providerNotFound)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        providerNotFound.ErrorRecord,
+                        providerNotFound));
+            }
+            catch (ItemNotFoundException pathNotFound)
+            {
+                WriteError(
+                    new ErrorRecord(
+                        pathNotFound.ErrorRecord,
+                        pathNotFound));
+            }
+
+            return results;
+        }
+
         /// <summary>
         /// Moves the specified item to the specified destination
         /// </summary>
         protected override void ProcessRecord()
+        {
+            if (base.SuppressWildcardExpansion)
+            {
+                RenameItem(Path, literalPath: true);
+                return;
+            }
+
+            Collection<PathInfo> resolvedPaths = GetResolvedPaths(Path);
+
+            switch (resolvedPaths.Count)
+            {
+                case 0:
+                    break;
+
+                case 1:
+                    string resolvedPath = resolvedPaths[0].Path;
+                    RenameItem(resolvedPath, literalPath: true);
+                    break;
+
+                default:
+                    RenameItem(Path, literalPath: true);
```

Comment:
Is this really working?

Reply:
I'd expect that we unescape path, check existance (or try rename), if fail then resolve globs (write an error if cpunt > 1) and try rename again.

Reply:
I believe -Path should try to resolve globs first, since it's the expected behaviour, and users can escape the wildcard character, for example
```PowerShell
Rename-Item -Path '`[A-B`]' -NewName 'A-B'
```
will prevent globbing.